package/MDAnalysis/analysis/align.py

```diff
@@ -1354,8 +1354,8 @@ def get_atoms_byres(g, match_mask=np.logical_not(mismatch_mask)):
                           "Try to improve your selections for mobile and reference.")
                 logger.error(errmsg)
                 raise SelectionError(errmsg)
-                
-    if match_atoms:
+
+    if match_atoms and hasattr(ag1, 'masses') and hasattr(ag2, 'masses'):
```

Comment:
Line where the code was changed

Reply:
It might be better if this line stays the same and a warning is issued when no mass can be found (potentially via another except).
The format will be similar to `warnings.warn(msg, category=SelectionWarning)`.
Please read the doc to check the types of warnings available and think of a meaningful message.